sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -406,7 +406,7 @@ object PreprocessTableInsertion extends Rule[LogicalPlan] {
       catalogTable.get.tracksPartitionsInCatalog
     if (partitionsTrackedByCatalog && normalizedPartSpec.nonEmpty) {
       // empty partition column value
-      if (normalizedPartSpec.map(_._2)
+      if (normalizedPartSpec.values
           .filter(_.isDefined).map(_.get).exists(v => v != null && v.isEmpty)) {
```

Comment:
How about just .flatten instead of the filter and map? I think that's another simplification that's valid.

Reply:
7eac600 fix this